Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -147,102 +191,211 @@ void FieldProbe::ComputeDiags (int step)
+                    amrex::ParticleReal xp, yp, zp;
+                    getPosition(ip, xp, yp, zp);
+
+                    amrex::ParticleReal Exp = 0._rt, Eyp = 0._rt, Ezp = 0._rt;
+                    amrex::ParticleReal Bxp = 0._rt, Byp = 0._rt, Bzp = 0._rt;
+                    amrex::ParticleReal S = 0._rt;
+
+                    // first gather E and B to the particle positions
+                    doGatherShapeN(xp, yp, zp, Exp, Eyp, Ezp, Bxp, Byp, Bzp,
+                                   arrEx, arrEy, arrEz, arrBx, arrBy, arrBz,
+                                   Extype, Eytype, Eztype, Bxtype, Bytype, Bztype,
+                                   dx_arr, xyzmin_arr, amrex::lbound(box), temp_modes,
+                                   temp_interp_order, false);
+
+                    //Calculate the Poynting Vector S
+                    amrex::Real const sraw[3]{
+                    Exp * Bzp - Ezp * Byp,
+                    Ezp * Bxp - Exp * Bzp,
+                    Exp * Byp - Eyp * Bxp};
+                    S = (1._rt / PhysConst::mu0)  * sqrt(sraw[0] * sraw[0] + sraw[1] * sraw[1] + sraw[2] * sraw[2]);
+
+                    /*
+                     * Determine whether or not to integrate field data.
+                     * If not integrating, store instantaneous values.
+                     */
+
+                    if (temp_field_probe_integrate != 1)
+                    {
+
+                        //Store Values on Particles
+
+                        part_Ex[ip] = Exp; //remember to add lorentz transform
+                        part_Ey[ip] = Eyp; //remember to add lorentz transform
+                        part_Ez[ip] = Ezp; //remember to add lorentz transform
+                        part_Bx[ip] = Bxp; //remember to add lorentz transform
+                        part_By[ip] = Byp; //remember to add lorentz transform
+                        part_Bz[ip] = Bzp; //remember to add lorentz transform
+                        part_S[ip] = S; //remember to add lorentz transform
+                    }
+                    else
+                    {
+                        part_Ex[ip] += Exp; //remember to add lorentz transform
+                        part_Ey[ip] += Eyp; //remember to add lorentz transform
+                        part_Ez[ip] += Ezp; //remember to add lorentz transform
+                        part_Bx[ip] += Bxp; //remember to add lorentz transform
+                        part_By[ip] += Byp; //remember to add lorentz transform
+                        part_Bz[ip] += Bzp; //remember to add lorentz transform
+                        part_S[ip] += S; //remember to add lorentz transform
+                    }
+                });// ParallelFor Close
+                if (field_probe_integrate == 0)
+                {
+                    for (int ip=0; ip < np; ip++)
+                    {
+                        // Fill output array
+                        m_data[0 * noutputs + ParticleVal::Ex] = part_Ex[ip];
```

Comment:
```suggestion
Requires managed memory without an explicit copy. Not a problem, but to be reworked later on.
```suggestion
                        // TODO replace with explicit copies later on to avoid managed memory on GPU
                        m_data[0 * noutputs + ParticleVal::Ex] = part_Ex[ip];
```